Controllers/ProductsController.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class ProductsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public ProductsController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Products
+        public async Task<IActionResult> Index()
+        {
+            var Products = await _context.Products
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .ToListAsync();
+
+            var branchesVM = new List<ProductFormViewModel>();
+
+            foreach (var product in Products)
+            {
+                branchesVM.Add(_mapper.Map<ProductFormViewModel>(product));
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Products by product Id
+        public async Task<IActionResult> ListProducts(int? id)
+        {
```

Comment:
Lack of null check against of nullable `id`. 
Remember nullable types expose dedicated interface for null check, so you can do both:
```c# 
if(id == null) {...}
//or
if(id.HasValue) {...}
```

Reply:
For branch without products, an empty is displayed as-is.

**TO BE IMPLEMENTED:** An warning message informing that the branch does not contain any products.